server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v71/CreateWebhooksTable.java

```diff
@@ -0,0 +1,126 @@
+import org.sonar.server.platform.db.migration.def.BigIntegerColumnDef;
+import org.sonar.server.platform.db.migration.def.VarcharColumnDef;
+import org.sonar.server.platform.db.migration.sql.CreateIndexBuilder;
+import org.sonar.server.platform.db.migration.sql.CreateTableBuilder;
+import org.sonar.server.platform.db.migration.step.DdlChange;
+
+import static org.sonar.server.platform.db.migration.def.BigIntegerColumnDef.newBigIntegerColumnDefBuilder;
+import static org.sonar.server.platform.db.migration.def.VarcharColumnDef.UUID_SIZE;
+import static org.sonar.server.platform.db.migration.def.VarcharColumnDef.newVarcharColumnDefBuilder;
+
+public class CreateWebhooksTable extends DdlChange {
+
+  private static final String TABLE_NAME = "webhooks";
+
+  private static final String ORGANIZATION_WEBHOOK_INDEX_NAME = "organization_webhook";
+  private static final String PROJECT_WEBHOOK_INDEX_NAME = "project_webhook";
+
+  private static final VarcharColumnDef UUID_COLUMN = newVarcharColumnDefBuilder()
+    .setColumnName("uuid")
+    .setIsNullable(false)
+    .setLimit(UUID_SIZE)
+    .build();
+  private static final int NAME_COLUMN_SIZE = 100;
+  private static final VarcharColumnDef NAME_COLUMN = newVarcharColumnDefBuilder()
+    .setColumnName("name")
+    .setIsNullable(false)
+    .setLimit(NAME_COLUMN_SIZE)
+    .build();
+  private static final int URL_COLUMN_SIZE = 2000;
+  private static final VarcharColumnDef URL_COLUMN = newVarcharColumnDefBuilder()
+    .setColumnName("url")
+    .setIsNullable(false)
+    .setLimit(URL_COLUMN_SIZE)
+    .build();
+  private static final VarcharColumnDef ORGANIZATION_UUID_COLUMN = newVarcharColumnDefBuilder()
+    .setColumnName("organization_uuid")
+    .setIsNullable(true)
+    .setLimit(UUID_SIZE)
+    .build();
+  private static final VarcharColumnDef PROJECT_UUID_COLUMN = newVarcharColumnDefBuilder()
+    .setColumnName("project_uuid")
+    .setIsNullable(true)
+    .setLimit(UUID_SIZE)
+    .build();
+  private static final BigIntegerColumnDef CREATED_AT_COLUMN = newBigIntegerColumnDefBuilder()
+    .setColumnName("created_at")
+    .setIsNullable(false)
+    .build();
+  private static final BigIntegerColumnDef UPDATED_AT_COLUMN = newBigIntegerColumnDefBuilder()
+    .setColumnName("updated_at")
+    .setIsNullable(true)
+    .build();
+
+  private Database db;
+
+  public CreateWebhooksTable(Database db) {
+    super(db);
+    this.db = db;
+  }
+
+  @Override
+  public void execute(Context context) throws SQLException {
+
+    if (!tableExists()) {
+      context.execute(new CreateTableBuilder(getDialect(), TABLE_NAME)
+        .addPkColumn(UUID_COLUMN)
+        .addColumn(NAME_COLUMN)
+        .addColumn(URL_COLUMN)
+        .addColumn(ORGANIZATION_UUID_COLUMN)
+        .addColumn(PROJECT_UUID_COLUMN)
+        .addColumn(CREATED_AT_COLUMN)
+        .addColumn(UPDATED_AT_COLUMN)
+        .build()
+      );
+
+      context.execute(new CreateIndexBuilder(getDialect())
```

Comment:
To be fully reentrant this creation of index should be ignored if index already exists. That can occur if failure occurs during the creation of next index.
That's not a problem for the time being. It's accepted to not be reentrant on creations of indices.

Reply:
ok.